File: Java/datepicker/app/src/main/java/com/example/date_picker/MainActivity.java
package com.example.date_picker;

import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.widget.Button;
import android.widget.DatePicker;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    DatePicker picker;
    Button displayDate;
    TextView textview1;

    @SuppressLint("SetTextI18n")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        textview1=(TextView)findViewById(R.id.textView1);
        picker=(DatePicker)findViewById(R.id.datePicker);
        displayDate=(Button)findViewById(R.id.button1);

        textview1.setText("Current Date: "+getCurrentDate());

        displayDate.setOnClickListener(view -> textview1.setText("Change Date: "+getCurrentDate()));
    }

    public String getCurrentDate(){
        return (picker.getMonth() + 1) + "/" + // month is 0 based
                picker.getDayOfMonth() + "/" +
                picker.getYear();
    }
}

File: Kotlin/alertdialog/app/src/main/java/com/example/alert_dialog/MainActivity.kt
package com.example.alert_dialog

import android.app.AlertDialog
import android.content.DialogInterface
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity


class MainActivity : AppCompatActivity() {
    private var closeButton: Button? = null
    private var builder: AlertDialog.Builder? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        closeButton = findViewById<View>(R.id.button) as Button
        builder = AlertDialog.Builder(this)
        closeButton!!.setOnClickListener {
            //Uncomment the below code to Set the message and title from the strings.xml file
            //builder.setMessage(R.string.dialog_message) .setTitle(R.string.dialog_title);

            //Setting message manually and performing action on button click
            builder!!.setMessage("Do you want to close this application ?")
                .setCancelable(false) //Lambda example
                .setPositiveButton(
                    "Yes"
                ) { _: DialogInterface?, _: Int ->
                    finish()
                    Toast.makeText(
                        applicationContext, "you choose yes action for alertbox",
                        Toast.LENGTH_SHORT
                    ).show()
                }
                .setNegativeButton(
                    "No"
                ) { dialog, _ -> //  Action for 'NO' Button
                    dialog.cancel()
                    Toast.makeText(
                        applicationContext, "you choose no action for alertbox",
                        Toast.LENGTH_SHORT
                    ).show()
                }
            //Creating dialog box
            val alert: AlertDialog = builder!!.create()
            //Setting the title manually
            alert.setTitle(R.string.dialog_title)
            alert.show()
        }
    }
}